library/oneview_unmanaged_device_facts.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/python
+
+###
+# Copyright (2016) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+
+from ansible.module_utils.basic import *
+from hpOneView.oneview_client import OneViewClient
+
+DOCUMENTATION = '''
+---
+module: oneview_unmanaged_device_facts
+short_description: Retrieve facts about one or more of the OneView Unmanaged Device.
+description:
+    - Retrieve facts about one or more of the Unmanaged Device from OneView.
+requirements:
+    - "python >= 2.7.9"
+    - "hpOneView"
+author: "Bruno Souza (@bsouza)"
+options:
+    config:
+      description:
+        - Path to a .json configuration file containing the OneView client configuration.
+      required: true
+    name:
+      description:
+        - Unmanaged Device name.
+      required: false
+    options:
+      description:
+        - "List with options to gather additional facts about Unmanaged Device.
+          Options allowed:
+          'environmental_configuration' gets a description of the environmental configuration for the Unmnaged Device."
+notes:
+    - "A sample configuration file for the config parameter can be found at:
+      https://github.hpe.com/Rainforest/oneview-ansible/blob/master/examples/oneview_config.json"
+'''
+
+EXAMPLES = '''
+- name: Gather facts about all Unmanaged Devices
+  oneview_unmanaged_device_facts:
+    config: "{{ config }}"
+
+- debug: var=unmanaged_devices
+
+- name: Gather facts about an Unmanaged Device by name
+  oneview_unmanaged_device_facts:
+    config: "{{ config }}"
+    name: "{{ name }}"
+
+- debug: var=unmanaged_devices
+
+- name: Gather facts about an Unmanaged Device by name with environmental configuration
+  oneview_unmanaged_device_facts:
+    config: "{{ config }}"
+    name: "{{ name }}"
+    options:
+      - environmental_configuration
+
+- debug: var=environmental_configuration
+'''
+
+RETURN = '''
+unmanaged_devices:
+    description: The list of unmanaged devices.
+    returned: Always, but can be null.
+    type: list
+
+environmental_configuration:
```

Comment:
I've seen that other modules are using the resource type as prefix on the name of the variable returned. For example, server_hardware_env_config, switch_environmental_configuration and enclosure_env_config (my only concern is that it don't follow a standard, sometimes is env_config, and sometimes is environmental_configuration).